src/runtime_src/core/tools/common/ReportDebugIpStatus.cpp

```diff
@@ -0,0 +1,1305 @@
+    amResults.CuMinExecCycles[index]  = ((uint64_t)(currData[5])) << 32;
+    amResults.CuMaxExecCycles[index]  = ((uint64_t)(currData[6])) << 32;
+    amResults.CuStartCount[index]     = ((uint64_t)(currData[7])) << 32;
+
+    if(hasDataflow) {
+      uint64_t dfTmp[2] = {0};
+#ifndef _WIN32
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
+      xclRead(handle, XCL_ADDR_SPACE_DEVICE_PERFMON, dbgIpInfo->m_base_address + XAM_BUSY_CYCLES_UPPER_OFFSET, &dfTmp[0], sizeof(uint32_t));
+      xclRead(handle, XCL_ADDR_SPACE_DEVICE_PERFMON, dbgIpInfo->m_base_address + XAM_MAX_PARALLEL_ITER_UPPER_OFFSET, &dfTmp[1], sizeof(uint32_t));
+#ifndef _WIN32
+#pragma GCC diagnostic pop
+#endif
+
+      amResults.CuBusyCycles[index]      = dfTmp[0] << 32;
+      amResults.CuMaxParallelIter[index] = dfTmp[1] << 32;
+    }
+  }
+
+  for (int c=0; c < XAM_DEBUG_SAMPLE_COUNTERS_PER_SLOT; c++) {
+#ifndef _WIN32
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
+    size += xclRead(handle, XCL_ADDR_SPACE_DEVICE_PERFMON, dbgIpInfo->m_base_address+am_offsets[c], &currData[c], sizeof(uint32_t));
+#ifndef _WIN32
+#pragma GCC diagnostic pop
+#endif
+  }
+
+  amResults.CuExecCount[index]      |= currData[0];
+  amResults.CuExecCycles[index]     |= currData[1];
+  amResults.CuStallExtCycles[index] |= currData[2];
+  amResults.CuStallIntCycles[index] |= currData[3];
+  amResults.CuStallStrCycles[index] |= currData[4];
+  amResults.CuMinExecCycles[index]  |= currData[5];
+  amResults.CuMaxExecCycles[index]  |= currData[6];
+  amResults.CuStartCount[index]     |= currData[7];
+
+  if(hasDataflow) {
+    uint64_t dfTmp[2] = {0};
+#ifndef _WIN32
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
+    xclRead(handle, XCL_ADDR_SPACE_DEVICE_PERFMON, dbgIpInfo->m_base_address + XAM_BUSY_CYCLES_OFFSET, &dfTmp[0], sizeof(uint32_t));
+    xclRead(handle, XCL_ADDR_SPACE_DEVICE_PERFMON, dbgIpInfo->m_base_address + XAM_MAX_PARALLEL_ITER_OFFSET, &dfTmp[1], sizeof(uint32_t));
+#ifndef _WIN32
+#pragma GCC diagnostic pop
+#endif
+
+    amResults.CuBusyCycles[index]      |= dfTmp[0] << 32;
+    amResults.CuMaxParallelIter[index] |= dfTmp[1] << 32;
+  } else {
+    amResults.CuBusyCycles[index]      = amResults.CuExecCycles[index];
+    amResults.CuMaxParallelIter[index] = 1;
+  }
+
+  (void)size;
+
+  // increment debugIpNum
+  ++debugIpNum[ACCEL_MONITOR];
+  amResults.NumSlots = (unsigned int)debugIpNum[ACCEL_MONITOR];
+}
+
+void 
+DebugIpStatusCollector::printAMResults(std::ostream& _output)
+{
+  if(0 == amResults.NumSlots) {
+    return;
+  }
+
+  _output << "\nAccelerator Monitor Counters (hex values are cycle count)\n";
+
+  auto col1 = std::max(cuNameMaxStrLen[ACCEL_MONITOR], strlen("Compute Unit")) + 4;
+
+  auto iosguard = xrt_core::utils::ios_restore(_output);
+  _output << std::left
```

Comment:
Instead of using the iostream formatting (which can get rather obfuscated), could you use boost::format.  For example:
```
boost::format header(std::to_string("%")+std::to_string(col1)+"s %8s  %16s  %16s  %16s  %16s  %16s  %16s  %16s);
_output << header % "Compute Unit"  % "Ends" % "Starts" % "Max Parallel Itr" % "Execution" % "Memory Stall" % "Pipe Stall" % "Stream Stall" % "Min Exec" % "Max Exec";

```
And the contents can be formatted similarly.